### PR TITLE
Run golangci-lint fmt in the format/lint targets

### PR DESCRIPTION
**Pull request by pkwarren**
Previously, formatters were run alongside linters when running `golangci-lint run`. In v2, they are configured and run separately with the `golangci-lint fmt` command.

Update the `format` target to run the formatter (if golangci-lint is not disabled). Update the `lint` target to run with `--diff` which will exit with a non-zero exit code if the code is not formatted.

This should ensure consistency with the behavior prior to upgrading to v2.

### CODE REVIEW

File: make/go/dep_golangci_lint.mk [hunk 1]
```diff
@@ -30,6 +30,7 @@ GOLANGCI_LINT_VERSION ?= v2.0.2
 GOLANGCI_LINT := $(CACHE_VERSIONS)/golangci-lint/$(GOLANGCI_LINT_VERSION)
 $(GOLANGCI_LINT):
 	@rm -f $(CACHE_BIN)/golangci-lint
+	@mkdir -p $(CACHE_BIN)
```
Comment (pkwarren) [inline make/go/dep_golangci_lint.mk:33]: Discovered this was missing while testing w/ an empty `~/.cache/makego`.

**Review by emcfarlane — APPROVED** [review]
